### PR TITLE
Add donation and other links to manifest

### PR DESCRIPTION
**Pull request by Madis0**
Compatible with [Mod Menu](https://github.com/TerraformersMC/ModMenu/wiki/API#links) and potentially anything else that might want to read it.

### CODE REVIEW

**Comment by douira** [comment]
Is this sufficient for FO to stop hiding Sodium's donation button by default?

**Comment by Madis0** [comment]
No, this is supplementary for any modpack or user who might hide the button, or any user who is looking to find the link from Mod Menu in the first place. I'll make a separate PR for FO concern.

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -24,7 +24,16 @@
     ]
   },
   "custom": {
-    "fabric-renderer-api-v1:contains_renderer": true
+    "fabric-renderer-api-v1:contains_renderer": true,
+    "modmenu": {
+      "links": {
+        "modmenu.donate": "https://jellysquid.me/donate",
```
Comment (jellysquid3) [inline src/main/resources/fabric.mod.json]: Should be `https://caffeinemc.net/donate` even if it just redirects currently. We may change it at some point.

File: src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -24,7 +24,16 @@
...
+        "modmenu.discord": "https://caffeinemc.net/discord",
+        "modmenu.github_releases": "https://github.com/CaffeineMC/sodium-fabric/releases",
+        "modmenu.kofi": "https://ko-fi.com/jellysquid_",
```
Comment (jellysquid3) [inline src/main/resources/fabric.mod.json]: Shouldn't directly link to Ko-Fi, see prior comment.

**Review by jellysquid3 — CHANGES_REQUESTED** [review]

**Review by jellysquid3 — APPROVED** [review]

**Pull request merged**